File: PHP/Fibonacci.php
<?php

$time_start = microtime(true); // start the timer

// fucntion to print the n fibonacci numbers

function printFibonacciNumbers($n)
{
    $f1 = 0;
    $f2 = 1; 
    $i;
 
    if ($n < 1)
        return;
    echo($f1);
    echo("<br>");
    for ($i = 1; $i < $n; $i++)
    {
        echo($f2);
        echo("<br>");
        $next = $f1 + $f2;
        $f1 = $f2;
        $f2 = $next;
    }
}

    printFibonacciNumbers(10001); // call the function to print the first 10000 fibonacci numbers

// Measurement

$memory_size = memory_get_usage(); // get the current memory usage

$memory_unit = array('Bytes','KB','MB','GB','TB','PB');

echo 'Used Memory : '.round($memory_size/pow(1024,($x=floor(log($memory_size,1024)))),2).' '.$memory_unit[$x]."<br>";

function win_sys_current_cpu_usage() { // get the current cpu usage
    $cmd = 'typeperf  -sc 1  "\Processor(_Total)\% Processor Time"';
    exec($cmd, $lines, $retval);
    if($retval == 0) {
      $values = str_getcsv($lines[2]);
      return floatval($values[1]);
    } else {
      return false;
    }
  }

echo 'Used CPU : ',win_sys_current_cpu_usage(),' %<br>';

$time_end = microtime(true); // end the timer
$execution_time = ($time_end - $time_start);
echo '<b>Total Execution Time:</b> '.$execution_time.' Seconds';

File: Python/Fibonacci.py
import os
import psutil
import time

def printFibonacciNumbers(n):
    
    # start the timer

    start_time = time.time()

    f1 = 0
    f2 = 1
    if (n < 1):
        return
    print(f1, end="\n")

    # for loop to print the n fibonacci numbers

    for x in range(1, n):
        print(f2, end="\n")
        next = f1 + f2
        f1 = f2
        f2 = next
     # get and print the runtime

    print("--- %s seconds ---" % (time.time() - start_time))
    
    # get and print the current cpu usage

    print("CPU : %s" % psutil.cpu_percent())
    
    # get the current memory usage

    process = psutil.Process(os.getpid())
    
    # print the current memory usage

    print("Memory : %s bytes" % process.memory_info().rss)

if __name__ == '__main__':

    # call the function to print the first 10000 fibonacci numbers

    printFibonacciNumbers(10001)
